lib/waterline/utils/query/private/normalize-constraint.js

```diff
@@ -44,6 +44,7 @@ var MODIFIER_KINDS = {
   'in':         true,
 
   'like':       true,
+  'ilike':      true,
   'contains':   true,
   'startsWith': true,
   'endsWith':   true
```

Comment:
The lay of the land right now is that the constraint modifiers like `like`, `in`, etc are (for now) standard, across all adapters, and currently, case-sensitivity of lookup behavior varies based on the database -- that was a Waterline 0.13 change we made to allow for better performance.  That did come at the cost of interoperability, and someday I'd like to solve that problem more elegantly.

@gstreetmedia But for the immediate term, I'm gathering you have a need to do case insensitive lookups on PostgreSQL, which totally makes sense, so focusing in on that for a sec:  Seems like there's two options:

1. Make `ilike` a formal part of the Waterline adapter spec and implement it across the board.  In some databases/charsets (like default MySQL), I think it'll be exactly the same thing as `like`.  A lot of the work for that is going to be in Waterline core.  It also involves updating docs, and possibly also waterline-utils.  We'd probably also want to update the other core adapters, including sails-mongo -- if not to support this, then at least to throw an error indicating that it's not supported in that adapter.   **(hard)**
2. Implement support for a `meta` key in sails-postgresql that makes all LIKE modifiers case-insensitive **(easy)**

My gut says that we should only bother messing with the second option right now.  It's less powerful, since it means you don't have granular control over the case-sensitivity of the various LIKE modifiers in the query (it's all or nothing).  This would only involve changing sails-postgresql, and I believe it could be done by recursively parsing the criteria ahead of time, transforming `like` into `ilike`

P.S. By the way, we shouldn't even have `startsWith`, `endsWith`, and `contains` in here -- they're [normalized ahead of time in Waterline core](https://github.com/balderdashy/waterline/blob/c146516fb625531717c5d7eda19cf8b99c1a6fb7/lib/waterline/utils/query/private/normalize-constraint.js#L609-L783).  I suspect that @particlebanana either added them back here because of a few places in the join logic where we were previously calling out directly to adapter methods without first forging the query.  That should be resolved now so we can probably remove them -- although they're not exactly hurting anything.